src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -1028,6 +1029,15 @@ export const GlobalProvider = memo(
                 });
                 outputDataActions.delete(id);
                 addInputDataChanges();
+                const nodeSchemaId = getNode(id)?.data.schemaId;
+                if (nodeSchemaId && isStartingNode(schemata.get(nodeSchemaId))) {
```

Comment:
I don't like the `isStartingNode` check here:

1. Even if the node isn't a starting node, then it won't do anything. We'd just send a small request to the backend. That's basically no cost.
2. I don't like that we cement the assumption that only starting nodes get executed here as well.